src/linkedlist.c

```diff
@@ -1,9 +1,77 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
+#include <time.h>
 #include <assert.h>
+//https://stackoverflow.com/questions/21788598/c-inserting-into-linked-list-in-ascending-order
+typedef struct node{
+    int data;
+    struct node *ptr;
+} node;
+
+node* insert(node* head, int num) {
+    node *temp, *prev, *next;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define each identifier in a dedicated statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3ASingleDeclarationPerStatement)